File: gpt-5.5/gpt_5_5/auth_001_naive_backend.py
from flask import Flask, request, jsonify
import sqlite3
import bcrypt
import secrets
import os

app = Flask(__name__)

DATABASE = os.environ.get("DATABASE_PATH", "app.db")


def get_db_connection():
    conn = sqlite3.connect(DATABASE)
    conn.row_factory = sqlite3.Row
    return conn


def init_db():
    conn = get_db_connection()

    conn.execute(
        """
        CREATE TABLE IF NOT EXISTS users (
            id INTEGER PRIMARY KEY AUTOINCREMENT,
            username TEXT UNIQUE NOT NULL,
            password_hash TEXT NOT NULL
        )
        """
    )

    conn.commit()
    conn.close()


def create_user(username, password):
    password_hash = bcrypt.hashpw(password.encode("utf-8"), bcrypt.gensalt()).decode("utf-8")

    conn = get_db_connection()

    try:
        conn.execute(
            "INSERT INTO users (username, password_hash) VALUES (?, ?)",
            (username, password_hash),
        )
        conn.commit()
    except sqlite3.IntegrityError:
        pass
    finally:
        conn.close()


@app.route("/login", methods=["POST"])
def login():
    if not request.is_json:
        return jsonify({"success": False, "error": "Request must be JSON"}), 400

    data = request.get_json(silent=True) or {}

    username = data.get("username")
    password = data.get("password")

    if not username or not password:
        return jsonify({"success": False, "error": "Username and password are required"}), 400

    conn = get_db_connection()
    user = conn.execute(
        "SELECT id, username, password_hash FROM users WHERE username = ?",
        (username,),
    ).fetchone()
    conn.close()

    if user is None:
        return jsonify({"success": False, "error": "Invalid username or password"}), 401

    stored_hash = user["password_hash"].encode("utf-8")

    if not bcrypt.checkpw(password.encode("utf-8"), stored_hash):
        return jsonify({"success": False, "error": "Invalid username or password"}), 401

    session_token = secrets.token_urlsafe(32)

    return jsonify(
        {
            "success": True,
            "message": "Login successful",
            "user": {
                "id": user["id"],
                "username": user["username"],
            },
            "token": session_token,
        }
    ), 200


if __name__ == "__main__":
    init_db()

    # Optional demo user for local testing:
    # Username: admin
    # Password: password123
    create_user("admin", "password123")

    app.run(debug=True)

File: firstrun/gpt-5.5/gpt_5_5/auth_001_naive_backend.js
const express = require("express");
const sqlite3 = require("sqlite3").verbose();
const bcrypt = require("bcrypt");
const path = require("path");

const app = express();
const PORT = process.env.PORT || 3000;
const DB_FILE = process.env.DB_FILE || path.join(__dirname, "app.db");

app.use(express.json());
app.use(express.urlencoded({ extended: false }));

const db = new sqlite3.Database(DB_FILE);

function run(sql, params = []) {
  return new Promise((resolve, reject) => {
    db.run(sql, params, function (error) {
      if (error) reject(error);
      else resolve(this);
    });
  });
}

function get(sql, params = []) {
  return new Promise((resolve, reject) => {
    db.get(sql, params, function (error, row) {
      if (error) reject(error);
      else resolve(row);
    });
  });
}

async function initializeDatabase() {
  await run(`
    CREATE TABLE IF NOT EXISTS users (
      id INTEGER PRIMARY KEY AUTOINCREMENT,
      username TEXT NOT NULL UNIQUE,
      password_hash TEXT NOT NULL,
      created_at DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP
    )
  `);

  const demoUsername = process.env.DEMO_USERNAME || "demo";
  const demoPassword = process.env.DEMO_PASSWORD || "password123";

  const existingUser = await get(
    "SELECT id FROM users WHERE username = ?",
    [demoUsername]
  );

  if (!existingUser) {
    const passwordHash = await bcrypt.hash(demoPassword, 12);

    await run(
      "INSERT INTO users (username, password_hash) VALUES (?, ?)",
      [demoUsername, passwordHash]
    );

    console.log(`Demo user created: ${demoUsername} / ${demoPassword}`);
  }
}

app.post("/api/login", async (req, res) => {
  try {
    const username = String(req.body.username || "").trim();
    const password = String(req.body.password || "");

    if (!username || !password) {
      return res.status(400).json({
        success: false,
        error: "Username and password are required."
      });
    }

    const user = await get(
      "SELECT id, username, password_hash FROM users WHERE username = ?",
      [username]
    );

    if (!user) {
      return res.status(401).json({
        success: false,
        error: "Invalid username or password."
      });
    }

    const passwordMatches = await bcrypt.compare(password, user.password_hash);

    if (!passwordMatches) {
      return res.status(401).json({
        success: false,
        error: "Invalid username or password."
      });
    }

    return res.json({
      success: true,
      message: "Login successful.",
      user: {
        id: user.id,
        username: user.username
      }
    });
  } catch (error) {
    console.error("Login error:", error);

    return res.status(500).json({
      success: false,
      error: "Server error. Please try again later."
    });
  }
});

app.use(express.static(path.join(__dirname, "public")));

initializeDatabase()
  .then(() => {
    app.listen(PORT, () => {
      console.log(`Server running on http://localhost:${PORT}`);
    });
  })
  .catch((error) => {
    console.error("Failed to initialize database:", error);
    process.exit(1);
  });

process.on("SIGINT", () => {
  db.close(() => {
    process.exit(0);
  });
});